contracts/stake/src/distribution.rs

```diff
@@ -1,32 +1,18 @@
-use soroban_sdk::{contracttype, Address, Env};
+use soroban_sdk::{contracttype, Address, Env, Vec};
 
 use curve::Curve;
 
 use crate::error::ContractError;
 
-#[contracttype]
-pub struct StorageCurve {
-    pub manager: Address,
-    pub start_timestamp: u64,
-    pub stop_timestamp: u64,
-    pub amount_to_distribute: u128,
-}
-
 // one reward distribution curve over one denom
-pub fn save_reward_curve(env: &Env, asset: &Address, distribution_curve: &StorageCurve) {
+pub fn save_reward_curve(env: &Env, asset: &Address, distribution_curve: &Curve) {
     env.storage().persistent().set(&asset, distribution_curve);
 }
 
 #[allow(dead_code)]
 pub fn get_reward_curve(env: &Env, asset: &Address) -> Result<Curve, ContractError> {
```

Comment:
is this function needed somewhere in the future?

Reply:
Yup. Since I'm saving the curve, obviously at some point I'm going to load it.